java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -0,0 +1,376 @@
+        // process block exist, which is conditional jump such as if-statement
+        checkerDispatcher.executeCheckPreStatement(programPosition.block.terminator());
+      } else {
+        // process branch
+        handleBlockExit(programPosition);
+      }
+    }
+
+    checkerDispatcher.executeCheckEndOfExecution();
+    // Cleanup:
+    explodedGraph = null;
+    workList = null;
+    node = null;
+    programState = null;
+    constraintManager = null;
+  }
+
+  private void handleBlockExit(ExplodedGraph.ProgramPoint programPosition) {
+    CFG.Block block = programPosition.block;
+    if (block.terminator() != null) {
+      switch (block.terminator().kind()) {
+        case IF_STATEMENT:
+          handleBranch(block, ((IfStatementTree) block.terminator()).condition());
+          return;
+        case CONDITIONAL_OR:
+        case CONDITIONAL_AND:
+          handleBranch(block, ((BinaryExpressionTree) block.terminator()).leftOperand());
+          return;
+        case CONDITIONAL_EXPRESSION:
+          handleBranch(block, ((ConditionalExpressionTree) block.terminator()).condition());
+          return;
+        case FOR_STATEMENT:
+          ForStatementTree forStatement = (ForStatementTree) block.terminator();
+          if (forStatement.condition() != null) {
+            handleBranch(block, forStatement.condition(), false);
+            return;
+          }
+      }
+    }
+    // unconditional jumps, for-statement, switch-statement:
+    for (CFG.Block successor : block.successors()) {
+      enqueue(new ExplodedGraph.ProgramPoint(successor, 0), programState);
+    }
+  }
+
+  private void handleBranch(CFG.Block programPosition, Tree condition) {
+    handleBranch(programPosition, condition, true);
+  }
+
+  private void handleBranch(CFG.Block programPosition, Tree condition, boolean checkPath) {
+    Pair<ProgramState, ProgramState> pair = constraintManager.assumeDual(programState, condition);
+    if (pair.a != null) {
+      // enqueue false-branch, if feasible
+      enqueue(new ExplodedGraph.ProgramPoint(programPosition.falseBlock(), 0), pair.a);
+      if (checkPath) {
+        alwaysTrueOrFalseCheck.evaluatedToFalse(condition);
+      }
+    } else {
+      SymbolicValue val = getVal(condition);
+      if (val != null) {
+        programState = ConstraintManager.setConstraint(programState, val, ConstraintManager.BooleanConstraint.TRUE);
+      }
+    }
+    if (pair.b != null) {
+      // enqueue true-branch, if feasible
+      enqueue(new ExplodedGraph.ProgramPoint(programPosition.trueBlock(), 0), pair.b);
+      if (checkPath) {
+        alwaysTrueOrFalseCheck.evaluatedToTrue(condition);
+      }
+    } else {
+      SymbolicValue val = getVal(condition);
+      if (val != null) {
+        programState = ConstraintManager.setConstraint(programState, val, ConstraintManager.BooleanConstraint.FALSE);
+      }
+    }
+
+  }
+
+  private void visit(Tree tree, Tree terminator) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "visit" is 18 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)